plugins/plugin-java-test-runner/che-java-testing-testng/che-java-testing-testng-ide/src/main/java/org/eclipse/che/ide/ext/java/testing/testng/client/action/RunTestXMLAction.java

```diff
@@ -0,0 +1,109 @@
+import org.eclipse.che.api.promises.client.OperationException;
+import org.eclipse.che.api.promises.client.Promise;
+import org.eclipse.che.api.promises.client.PromiseError;
+import org.eclipse.che.ide.api.action.ActionEvent;
+import org.eclipse.che.ide.api.editor.EditorAgent;
+import org.eclipse.che.ide.api.filetypes.FileTypeRegistry;
+import org.eclipse.che.ide.api.notification.NotificationManager;
+import org.eclipse.che.ide.api.notification.StatusNotification;
+import org.eclipse.che.ide.api.resources.Project;
+import org.eclipse.che.ide.ext.java.client.action.JavaEditorAction;
+import org.eclipse.che.ide.ext.java.testing.core.client.TestServiceClient;
+import org.eclipse.che.ide.ext.java.testing.core.client.view.TestResultPresenter;
+import org.eclipse.che.ide.ext.java.testing.core.shared.TestResult;
+import org.eclipse.che.ide.ext.java.testing.testng.client.TestNGLocalizationConstant;
+import org.eclipse.che.ide.ext.java.testing.testng.client.TestNGResources;
+import org.eclipse.che.ide.rest.DtoUnmarshallerFactory;
+import org.eclipse.che.ide.util.loging.Log;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.eclipse.che.ide.api.notification.StatusNotification.DisplayMode.FLOAT_MODE;
+import static org.eclipse.che.ide.api.notification.StatusNotification.Status.*;
+
+/**
+ *
+ * @author Mirage Abeysekara
+ */
+public class RunTestXMLAction extends JavaEditorAction {
+
+    private final NotificationManager notificationManager;
+    private final EditorAgent editorAgent;
+    private TestResultPresenter presenter;
+    private final TestServiceClient service;
+    private final DtoUnmarshallerFactory dtoUnmarshallerFactory;
+
+    @Inject
+    public RunTestXMLAction(TestNGResources resources, NotificationManager notificationManager, EditorAgent editorAgent,
+                            TestResultPresenter presenter, FileTypeRegistry fileTypeRegistry, TestServiceClient service,
+                            DtoUnmarshallerFactory dtoUnmarshallerFactory, TestNGLocalizationConstant localization) {
+
+        super(localization.actionRunXMLTitle(), localization.actionRunXMLDescription(), resources.testAllIcon(),
+                editorAgent, fileTypeRegistry);
+
+        this.notificationManager = notificationManager;
+        this.editorAgent = editorAgent;
+        this.presenter = presenter;
+        this.service = service;
+        this.dtoUnmarshallerFactory = dtoUnmarshallerFactory;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        final StatusNotification notification = new StatusNotification("Running Tests...", PROGRESS, FLOAT_MODE);
+        notificationManager.notify(notification);
+
+        final Project project = appContext.getRootProject();
+
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("updateClasspath", "true");
+        parameters.put("testngXML", project.getPath() + "/src/test/resources/testng.xml");
```

Comment:
@vparfonov Do we have a way to figure out path to test resource if they are configured differently? I mean path is different.